### PR TITLE
Changes for using the adaptive hole map algorithm

### PR DESCRIPTION
**Pull request by itopcuoglu**

### CODE REVIEW

File: src/TiogaSTKIface.cpp [hunk 1]
```diff
@@ -56,6 +56,10 @@ void TiogaSTKIface::load(const YAML::Node& node)
     if (node["use_ngp_interface"]) {
         use_ngp_iface_ = node["use_ngp_interface"].as<bool>();
     }
+    if (node["use_adaptive_hole_map"]) {
+        use_adaptive_hole_map_ = node["use_adaptive_hole_map"].as<bool>();
+        if (use_adaptive_hole_map_ == true) tg_.setHoleMapAlgorithm(1);
```
Comment (marchdf) [inline src/TiogaSTKIface.cpp]: no need for `== true`

File: src/ExaTioga.cpp [hunk 2]
```diff
@@ -115,6 +115,7 @@ void ExaTioga::perform_connectivity()
     {
         auto tmon = tioga_nalu::get_timer("tioga::performConnectivity");
         amrex::Print() << "TIOGA unstructured connectivity" << std::endl;
+	m_tioga.assembleComms();
```
Comment (ashesh2512) [inline src/ExaTioga.cpp:118]: We should move this to after this - https://github.com/Exawind/tioga_utils/blob/exawind/src/ExaTioga.cpp#L40

Comment (marchdf) [reply src/ExaTioga.cpp:118]: I think (?) `assembleComms` should be called _in addition_ to `setCommunicator`. They look like they are doing very different things.

Comment (ashesh2512) [reply src/ExaTioga.cpp:118]: that is correct

Comment (itopcuoglu) [reply src/ExaTioga.cpp:118]: `setCommunicator()` basically creates two parallelComm instances, while `assembleComms()` processes meshblock information and then creates sub comms. The meshblock information that is used by `assembleComms()` seems to be created inside the Tioga function `register_unstructured_grid()`, which is called  by `TiogaBlock::register_block_ngp()`. So the `assembleComms()` call quite possibly has to happen after that is called.

**Pull request closed without merging**